win32ss/user/ntuser/winpos.c

```diff
@@ -2101,7 +2101,11 @@ co_WinPosSetWindowPos(
 
       /* We need to redraw what wasn't visible before or force a redraw */
       if ((WinPos.flags & (SWP_FRAMECHANGED | SWP_SHOWWINDOW)) ||
-          (((WinPos.flags & SWP_AGG_NOGEOMETRYCHANGE) != SWP_AGG_NOGEOMETRYCHANGE) && VisAfter != NULL))
+          ((((WinPos.flags & SWP_AGG_NOGEOMETRYCHANGE) != SWP_AGG_NOGEOMETRYCHANGE) ||
+          /* This fixes CORE-18830 */
+          ((Window->style & (WS_CHILD | WS_VISIBLE | WS_CLIPSIBLINGS | WS_CLIPCHILDREN)) ==
+          (WS_CHILD | WS_VISIBLE | WS_CLIPSIBLINGS | WS_CLIPCHILDREN)))
+          && VisAfter != NULL))
```

Comment:
That condition is getting more beautiful by the day. :-)
I doubt we will come to a correct solution ever by extending it. But I see no big risk in experimenting further with that small hack, as it is very local, and does exist already. But I do fear a bit, it may turn into a neverending story.
Approved out of curiosity! :-)

Reply:
I think that this is somewhat HACKish myself, but maybe if we get enough conditions, the real solution will become obvious. I am also looking into trying to simplify this. It seems that mixing 'WinPos.flags' and 'Window->style' may be redundant.

Reply:
Testbot results do look fairly good.

[CORE-18830](https://jira.reactos.org/browse/CORE-18830).patch JID65155 on top of 0.4.15-dev-5677-gf691efe

VBox: https://reactos.org/testman/compare.php?ids=85971,85973 LGTM
KVM:  https://reactos.org/testman/compare.php?ids=85970,85972 LGTM

Reply:
> maybe if we get enough conditions, the real solution will become obvious

That is 100% the hope I did base approval upon.